Sprint 2

AleksandrMenshchikov opened the pull request.

AleksandrMenshchikov committed:
new file:   .editorconfig

AleksandrMenshchikov committed:
new file:   utils/HTTPTransport.js

AleksandrMenshchikov committed:
new file:   utils/fetchWithRetry.js

AleksandrMenshchikov committed:
renamed:    app.js -> app.ts
	modified:   package-lock.json
	modified:   package.json
	new file:   tsconfig.json
	deleted:    utils/fetchWithRetry.js
	new file:   utils/fetchWithRetry.ts
	renamed:    utils/firstElementArray.js -> utils/firstElementArray.ts

AleksandrMenshchikov committed:
renamed:    utils/isEmpty.js -> utils/isEmpty.ts

AleksandrMenshchikov committed:
renamed:    utils/lastElementArray.js -> utils/lastElementArray.ts

AleksandrMenshchikov committed:
modified:   .editorconfig

AleksandrMenshchikov committed:
new file:   .eslintignore
	new file:   .eslintrc.json

AleksandrMenshchikov committed:
modified:   utils/fetchWithRetry.ts

AleksandrMenshchikov committed:
modified:   utils/firstElementArray.ts

AleksandrMenshchikov committed:
modified:   .eslintrc.json
	modified:   app.ts
	modified:   package-lock.json
	modified:   package.json
	deleted:    utils/HTTPTransport.js
	new file:   utils/HTTPTransport.ts
	modified:   utils/fetchWithRetry.ts
	modified:   utils/firstElementArray.ts
	modified:   utils/isEmpty.ts
	modified:   utils/lastElementArray.ts
	deleted:    utils/range.js
	new file:   utils/range.ts
	deleted:    utils/rangeRight.js
	new file:   utils/rangeRight.ts

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   assets/fonts/fonts.css
	modified:   package.json

AleksandrMenshchikov committed:
modified:   src/pages/500/index.css

AleksandrMenshchikov committed:
modified:   .eslintignore
	modified:   .gitignore
	new file:   .husky/pre-commit
	new file:   .stylelintrc.json
	modified:   package-lock.json
	modified:   package.json
	modified:   src/pages/404/index.css
	modified:   src/pages/404/index.hbs
	modified:   src/pages/chat/index.css
	modified:   src/pages/index.css
	modified:   src/pages/index.hbs
	modified:   src/pages/login/index.css
	modified:   src/pages/signup/index.css

AleksandrMenshchikov committed:
modified:   .eslintrc.json
	new file:   lib/eventBus.ts
	modified:   package-lock.json
	modified:   package.json

AleksandrMenshchikov committed:
renamed:    lib/eventBus.ts -> utils/eventBus.ts

AleksandrMenshchikov committed:
modified:   package-lock.json
	modified:   package.json

AleksandrMenshchikov committed:
modified:   .eslintignore
	new file:   .eslintrc
	deleted:    .eslintrc.json
	new file:   .npmrc
	new file:   .nvmrc
	modified:   .parcelrc
	modified:   README.md
	deleted:    handlebars.config.js
	modified:   package-lock.json
	modified:   package.json
	new file:   src/core/Block.ts
	new file:   src/core/EventBus.ts
	new file:   src/core/index.ts
	new file:   src/core/registerComponent.ts
	new file:   src/core/renderDOM.ts
	renamed:    src/pages/index.hbs -> src/pages/index.html
	new file:   src/pages/login/index.html
	new file:   src/pages/login/index.ts
	modified:   tsconfig.json
	new file:   typings/app.d.ts
	new file:   typings/hbs.d.ts
	deleted:    utils/eventBus.ts

AleksandrMenshchikov committed:
modified:   .eslintrc
	modified:   app.ts
	modified:   package.json

AleksandrMenshchikov committed:
modified:   src/core/Block.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   package-lock.json

AleksandrMenshchikov committed:
modified:   package-lock.json
	modified:   package.json

AleksandrMenshchikov committed:
modified:   src/pages/login/index.css

AleksandrMenshchikov committed:
modified:   src/pages/login/index.css

AleksandrMenshchikov committed:
new file:   src/components/input-login/index.ts
	new file:   src/components/input-login/input-login.css
	new file:   src/components/input-login/input-login.hbs
	new file:   src/components/input-login/input-login.ts
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
renamed:    .stylelintrc.json -> .stylelintrc
	modified:   package-lock.json

AleksandrMenshchikov committed:
modified:   README.md
	new file:   src/components/input-error/index.ts
	new file:   src/components/input-error/input-error.css
	new file:   src/components/input-error/input-error.hbs
	new file:   src/components/input-error/input-error.ts
	deleted:    src/components/input-login/index.ts
	deleted:    src/components/input-login/input-login.hbs
	deleted:    src/components/input-login/input-login.ts
	new file:   src/components/input/index.ts
	renamed:    src/components/input-login/input-login.css -> src/components/input/input.css
	new file:   src/components/input/input.hbs
	new file:   src/components/input/input.ts
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
new file:   src/components/button/button.css
	new file:   src/components/button/button.hbs
	new file:   src/components/button/button.ts
	new file:   src/components/button/index.ts
	modified:   src/components/input-error/input-error.css
	modified:   src/components/input-error/input-error.hbs
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   tsconfig.json

AleksandrMenshchikov committed:
deleted:    typings/app.d.ts

AleksandrMenshchikov committed:
modified:   .parcelrc
	modified:   package-lock.json
	modified:   package.json
	modified:   src/components/button/button.ts
	modified:   src/components/input-error/input-error.ts
	modified:   src/components/input/input.ts
	modified:   src/core/Block.ts
	modified:   src/core/EventBus.ts
	modified:   src/core/index.ts
	deleted:    src/core/registerComponent.ts
	modified:   src/core/renderDOM.ts
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input/input.ts
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   package.json
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input-error/input-error.ts
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input-error/input-error.hbs
	modified:   src/core/Block.ts
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/index.html
	modified:   src/pages/login/index.html
	modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.hbs
	new file:   src/pages/signup/index.html
	new file:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input-error/input-error.css
	modified:   src/pages/signup/index.hbs
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.hbs
	modified:   src/pages/signup/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/signup/index.hbs
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
deleted:    src/pages/404/index.hbs
	new file:   src/pages/404/index.html
	deleted:    src/pages/500/index.hbs
	new file:   src/pages/500/index.html
	renamed:    src/pages/chat/index.hbs -> src/pages/chat/index.html
	modified:   src/pages/index.html

AleksandrMenshchikov committed:
deleted:    src/layouts/base.hbs
	modified:   src/pages/404/index.html
	modified:   src/pages/500/index.html
	modified:   src/pages/chat/index.html
	modified:   src/pages/login/index.html
	modified:   src/pages/signup/index.html
	deleted:    src/partials/chat/chat.hbs

AleksandrMenshchikov committed:
modified:   package-lock.json
	modified:   package.json

AleksandrMenshchikov committed:
new file:   src/pages/404/index.hbs
	modified:   src/pages/404/index.html
	new file:   src/pages/404/index.ts
	new file:   src/pages/500/index.hbs
	modified:   src/pages/500/index.html
	new file:   src/pages/500/index.ts
	new file:   src/pages/chat/index.hbs
	modified:   src/pages/chat/index.html
	new file:   src/pages/chat/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/404/index.html
	modified:   src/pages/500/index.html
	modified:   src/pages/chat/index.html

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.hbs
	modified:   src/pages/chat/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs
	modified:   src/pages/chat/index.ts
	modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.hbs
	modified:   src/pages/chat/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/404/index.ts
	modified:   src/pages/500/index.ts
	modified:   src/pages/chat/index.html
	modified:   src/pages/chat/index.ts
	modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs
	modified:   src/pages/chat/index.ts

AleksandrMenshchikov committed:
renamed:    src/pages/chat/index.css -> src/pages/messenger/index.css
	renamed:    src/pages/chat/index.hbs -> src/pages/messenger/index.hbs
	renamed:    src/pages/chat/index.html -> src/pages/messenger/index.html
	renamed:    src/pages/chat/index.ts -> src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/index.html

AleksandrMenshchikov committed:
modified:   src/pages/404/index.hbs
	modified:   src/pages/500/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/index.html

AleksandrMenshchikov committed:
modified:   src/components/button/button.ts
	new file:   src/components/clip/clip.css
	new file:   src/components/clip/clip.hbs
	new file:   src/components/clip/clip.ts
	new file:   src/components/clip/index.ts
	modified:   src/components/input-error/input-error.ts
	modified:   src/components/input/input.ts
	modified:   src/components/member/member.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs

AleksandrMenshchikov committed:
new file:   src/components/dots/dots.css
	new file:   src/components/dots/dots.hbs
	new file:   src/components/dots/dots.ts
	new file:   src/components/dots/index.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/button-profile/button-profile.css
	new file:   src/components/button-profile/button-profile.hbs
	new file:   src/components/button-profile/button-profile.ts
	new file:   src/components/button-profile/index.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.hbs

AleksandrMenshchikov committed:
new file:   src/components/button-exit/button-exit.css
	new file:   src/components/button-exit/button-exit.hbs
	new file:   src/components/button-exit/button-exit.ts
	new file:   src/components/button-exit/index.ts
	modified:   src/components/button-profile/button-profile.css
	modified:   src/components/button/button.css
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input-error/input-error.css
	modified:   src/components/input/input.css
	modified:   src/components/input/input.hbs
	modified:   src/components/input/input.ts
	modified:   src/pages/login/index.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

gennady-bars commented on src/pages/index.html:
```diff
@@ -0,0 +1,23 @@
+  </head>
+  <body>
+    <main>
+      <a href="./login/index.html">Login</a>
```
Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту.

gennady-bars replied:
Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет.

gennady-bars commented on src/pages/index.html:
```diff
@@ -0,0 +1,23 @@
+      <a href="./login/index.html">Login</a>
+      <div><br /></div>
+      <a href="./signup/index.html">Signup</a>
+      <div><br /></div>
```
` <div><br /></div>`

не стоит использовать теги `div` и `br` для создания отступов. Это очень плохо по семантике


когда сделаете `ul` и `li`, то они сами отделятся

Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/pages/login/index.hbs:
```diff
@@ -1,38 +1,19 @@
     </form>
   </div>
-</main>
+</main>
```
В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/core/Block.ts:
```diff
@@ -0,0 +1,202 @@
+import { nanoid } from 'nanoid';
+import EventBus from './EventBus';
+
+export default class Block {
```
**Можно лучше**

лучше всего передавать тип пропсов в виде дженерика в `Block`, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Block`. Будет очень универсально 

```js
abstract class Block<Props extends Record<string, any> = unknown> {

```

gennady-bars commented on src/core/Block.ts:
```diff
@@ -0,0 +1,202 @@
+    FLOW_CDM: 'flow:component-did-mount',
+    FLOW_CDU: 'flow:component-did-update',
+    FLOW_RENDER: 'flow:render',
+  };
```
тут нужно добавить `as const`, чтобы зафиксировать литеральный тип строк


Подробнее https://igorfonin.gitbook.io/typescript-book-ru/typescript-type-system/literals 

```js

  } as const;

```

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/core/Block.ts:
```diff
@@ -0,0 +1,202 @@
+
+  private _meta: { props: any; };
+
+  protected props: any;
```
`  protected props: any;
`
Тип **пропсов** внутри класса `Block` должен быть единым. Либо он должен приходить в дженерик (смотрите мой совет), либо нужно объявить его 1 раз снаружи и использовать везде

gennady-bars commented on src/core/Block.ts:
```diff
@@ -0,0 +1,202 @@
+  initChildren() {}
+
+  // eslint-disable-next-line class-methods-use-this
+  _getChildren(propsAndChildren: any) {
```
очень много типов `any` в классе `Block`. Этого не должно быть. Это главный файл проекта

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/components/input/input.ts:
```diff
@@ -0,0 +1,31 @@
+    id, classValue, type, name, autocomplete, minLength, maxLength, pattern, events,
+  }:InputProps) {
+    super({
+      id, classValue, type, name, autocomplete, minLength, maxLength, pattern, events,
```
`      id, classValue, type, name, autocomplete, minLength, maxLength, pattern, events,
`

Вы не изменяете никак пропсы, значит, и не нужно деструктурировать объект


```js


  constructor(props :InputProps) {
    super(props);
  }

```

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/pages/login/index.ts:
```diff
@@ -0,0 +1,92 @@
+  constructor() {
+    super();
+    this.form = this.element.querySelector('form') as HTMLFormElement;
+    this.form.addEventListener('submit', (e) => {
```
`addEventListener`

любой **обработчик события** должен навешиваться только через **пропсы** экземпляра класса, который наследуется от `Block`. В теории показано было, что в пропсах передается объект `events` со всеми нужными событиями, а в родительском классе `Block` они навешиваются через `_addEvents` и удаляются через `_removeEvents`. Посмотрите реализацию `Block` в теории

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/pages/messenger/index.ts:
```diff
@@ -0,0 +1,366 @@
+      pattern: '^[a-zA-Z]+[a-zA-Z0-9-._]*@[a-zA-Z]+\\.[a-zA-Z]+$',
+      events: {
+        focus: (e) => {
+          this.handleInputFocusBlur(e, 'email');
```
**Можно лучше**


очень много раз дублируется `handleInputFocusBlur`. Вам нужно было получать данные атрибута `name` внутри этой функции, чтобы не передавать строку после `e`, тогда можно было бы просто 1 раз указать объект `events` внутри класса `Input`, вставляя его в `props`. Тогда Вы избежите 16ти дублирований (а точнее 32 раза)


атрибут `name` можно получить из объекта события вот так: `e.target.name` (это и есть то, что указано в `name`)

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/pages/messenger/index.ts:
```diff
@@ -0,0 +1,366 @@
+    e: Event,
+    element: 'login' | 'email' | 'first_name' | 'second_name' | 'phone' | 'password' | 'password-confirm' | 'password-old',
+  ) {
+    const isValidInput = (e.currentTarget as HTMLInputElement).checkValidity();
```
**Можно лучше**


надо вот так переделать эту функцию, чтобы она стала универсальной. Это избавит Вас от дублирований и передачи 2го параметра в вызов.

```js

  handleInputFocusBlur(
    e: Event
  ) {
// это и есть инпут
    const input = e.currentTarget as HTMLInputElement
    const isValidInput = input.checkValidity();
    if (!isValidInput) {
      input.parentElement?.nextElementSibling?.classList.add('error_active');
    } else {
      input.parentElement?.nextElementSibling?.classList.remove('error_active');
    }
  }


```

AleksandrMenshchikov replied:
Исправил.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что фиксируете версии зависимостей
* Хорошая структура папок и файлов
* Добавлен `ESLint` и `Stylelint`
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что не забываете про тег `main` для главной части страницы
* Отличные сообщения об ошибках под инпутами
* Отличное плавное появление деталей профиля


но есть некоторые недочеты: 


*  Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту.
* Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет.
* не стоит использовать теги `div` и `br` для создания отступов. Это очень плохо по семантике
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`
* Тип **пропсов** внутри класса `Block` должен быть единым. Либо он должен приходить в дженерик (смотрите мой совет), либо нужно объявить его 1 раз снаружи и использовать везде
* очень много типов `any` в классе `Block`. Этого не должно быть. Это главный файл проекта
* любой **обработчик события** должен навешиваться только через **пропсы** экземпляра класса, который наследуется от `Block`. В теории показано было, что в пропсах передается объект `events` со всеми нужными событиями, а в родительском классе `Block` они навешиваются через `_addEvents` и удаляются через `_removeEvents`. 


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

AleksandrMenshchikov committed:
modified:   src/pages/index.css
	modified:   src/pages/index.html
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/button-exit/button-exit.hbs
	modified:   src/components/button-profile/button-profile.hbs
	modified:   src/components/button/button.hbs
	modified:   src/components/clip/clip.hbs
	modified:   src/components/dots/dots.hbs
	modified:   src/components/input-error/input-error.hbs
	modified:   src/components/input/input.hbs
	modified:   src/components/member/member.hbs
	modified:   src/pages/404/index.hbs
	modified:   src/pages/500/index.hbs
	modified:   src/pages/login/index.hbs
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/signup/index.hbs
	modified:   src/pages/signup/index.html
	modified:   tsconfig.json

AleksandrMenshchikov committed:
modified:   src/core/Block.ts

AleksandrMenshchikov committed:
modified:   src/components/button-exit/button-exit.ts
	modified:   src/components/button-profile/button-profile.ts
	modified:   src/components/button/button.ts
	modified:   src/components/input-error/input-error.ts
	modified:   src/components/input/input.ts
	modified:   src/core/Block.ts
	modified:   src/core/EventBus.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/login/index.ts
	modified:   src/pages/messenger/index.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
new file:   src/components/form-login/form-login.css
	new file:   src/components/form-login/form-login.hbs
	new file:   src/components/form-login/form-login.ts
	new file:   src/components/form-login/index.ts
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
new file:   src/components/form-signup/form-signup.css
	new file:   src/components/form-signup/form-signup.hbs
	new file:   src/components/form-signup/form-signup.ts
	new file:   src/components/form-signup/index.ts
	modified:   src/pages/signup/index.css
	modified:   src/pages/signup/index.hbs
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
new file:   src/components/form-messenger/form-messenger.css
	new file:   src/components/form-messenger/form-messenger.hbs
	new file:   src/components/form-messenger/form-messenger.ts
	new file:   src/components/form-messenger/index.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/button-avatar/button-avatar.css
	new file:   src/components/button-avatar/button-avatar.hbs
	new file:   src/components/button-avatar/button-avatar.ts
	new file:   src/components/button-avatar/index.ts
	new file:   src/components/button-open-profile/button-open-profile.css
	new file:   src/components/button-open-profile/button-open-profile.hbs
	new file:   src/components/button-open-profile/button-open-profile.ts
	new file:   src/components/button-open-profile/index.ts
	modified:   src/components/form-messenger/form-messenger.css
	modified:   src/components/form-messenger/form-messenger.hbs
	modified:   src/components/form-messenger/form-messenger.ts
	new file:   src/components/profile-arrow/index.ts
	new file:   src/components/profile-arrow/profile-arrow.css
	new file:   src/components/profile-arrow/profile-arrow.hbs
	new file:   src/components/profile-arrow/profile-arrow.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
